be/src/vec/data_types/data_type_number_base.cpp

```diff
@@ -65,30 +65,40 @@ std::string DataTypeNumberBase<T>::to_string(const IColumn& column, size_t row_n
     }
 }
 
+// binary: column num | value1 | value2 | ...
 template <typename T>
-size_t DataTypeNumberBase<T>::serialize(const IColumn& column, PColumn* pcolumn) const {
-    const auto column_len = column.size();
-    pcolumn->mutable_binary()->resize(column_len * sizeof(FieldType));
-    auto* data = pcolumn->mutable_binary()->data();
+int64_t DataTypeNumberBase<T>::get_uncompressed_serialized_bytes(const IColumn& column) const {
+    return sizeof(uint32_t) + column.size() * sizeof(FieldType);
+}
 
-    // copy the data
+template <typename T>
+char* DataTypeNumberBase<T>::serialize(const IColumn& column, char* buf) const {
+    // column num
+    const auto column_num = column.size();
```

Comment:
Why each column need to serialize column size，one block only need register block->rows() once

Reply:
`IColumn + DataType` to `serialized data + PColumnData`. So the `column size` info is saved in `serialized data`